osquery/tables/system/darwin/signature.mm

```diff
@@ -0,0 +1,126 @@
+/*
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/filesystem.hpp>
+#include <Cocoa/Cocoa.h>                /* For NSAppKitVersionNumber */
+#include <Foundation/Foundation.h>
+#include <Security/CodeSigning.h>
+
+#include <osquery/core.h>
+#include <osquery/core/conversions.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+void genSignatureForFile(const std::string& path,
+                         QueryData& results) {
+  Row r;
+  OSStatus result = 0;
+  CFStringRef ident = NULL;
```

Comment:
You could delay the instantiation of these pointers/const pointers until later in the method. When you do initialize them, prefer to use `nullptr`, it will make worlds of different when we run our static analysis checks!